### PR TITLE
DTSRD-3404: Upgrade to spring boot 3

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by rapolaskaseliscgi — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -233,22 +209,25 @@ repositories {
...
-  springBoot          : '2.7.18',
-  springfoxSwagger    : '2.10.5',
-  serenity            : '2.0.76',
+  springBoot          : '3.3.4',
+  restAssured         : '5.5.0',
+  serenity            : '4.2.6',
   lombok              : '1.18.34',
-  flyway              : '6.5.7',
-  camel               : '3.8.0',
+  flyway              : '9.19.4',
```
Comment (lukasz-wolski) [inline build.gradle:224]: please update to 9.22.3, same version as in plugins section

File: build.gradle [hunk 2]
```diff
@@ -306,69 +271,43 @@ dependencies {
   implementation group: 'com.fasterxml.jackson', name: 'jackson-bom', version: '2.17.2', ext: 'pom'
   implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: versions.jackson
 
-  implementation group: 'org.yaml', name: 'snakeyaml', version: '1.33'
+  implementation group: 'org.yaml', name: 'snakeyaml', version: '2.3'
 
-  implementation ('com.github.hmcts:data-ingestion-lib:0.5.2.6') {
+  implementation ('com.github.hmcts:data-ingestion-lib:0.6.0.0-beta-13') {
```
Comment (lukasz-wolski) [inline build.gradle:276]: beta-13 still uses spring-boot 3.3.2, possibly use beta-14

**Review by lukasz-wolski — APPROVED** [review]
Please see comments inlined

**Review by lukasz-wolski — DISMISSED** [review]

**Review by SabinaHMCTS — DISMISSED** [review]

**Review by SabinaHMCTS — APPROVED** [review]

**Review by lukasz-wolski — APPROVED** [review]

File: src/main/resources/application-lrd-router.yaml [hunk 3]
```diff
@@ -12,9 +12,9 @@ route:
     insert-sql:
       sql:INSERT INTO service_to_ccd_case_type_assoc (service_code, ccd_service_name, ccd_case_type, created_date)
       VALUES (:#service_code,:#ccd_service_name,:#ccd_case_type, NOW() AT TIME ZONE 'utc')
-      ON CONFLICT (service_code, ccd_case_type) DO NOTHING ?batch=true&dataSource=dataSource
+      ON CONFLICT (service_code, ccd_case_type) DO NOTHING ?dataSource=#dataSource
```
Comment (rajkatla-hmcts) [inline src/main/resources/application-lrd-router.yaml:15]: May I know why batch mode has been disabled?

File: src/main/resources/application-lrd-building-location-router.yaml [hunk 4]
```diff
@@ -11,9 +11,9 @@ route:
...
       ON CONFLICT (epimms_id) DO UPDATE SET building_location_name = :#building_location_name,building_location_status = :#building_location_status,area = :#area,region_id = :#region_id,cluster_id = :#cluster_id,
-      court_finder_url = :#court_finder_url,postcode = :#postcode,address = :#address,welsh_building_location_name = :#welsh_building_location_name,welsh_address = :#welsh_address,uprn = :#uprn,latitude = :#latitude,longitude = :#longitude,mrd_building_location_id = :#mrd_building_location_id,mrd_created_time = :#mrd_created_time,mrd_updated_time = :#mrd_updated_time,mrd_deleted_time = :#mrd_deleted_time,updated_time = NOW() AT TIME ZONE 'utc'?batch=true&dataSource=dataSource
+      court_finder_url = :#court_finder_url,postcode = :#postcode,address = :#address,welsh_building_location_name = :#welsh_building_location_name,welsh_address = :#welsh_address,uprn = :#uprn,latitude = :#latitude,longitude = :#longitude,mrd_building_location_id = :#mrd_building_location_id,mrd_created_time = :#mrd_created_time,mrd_updated_time = :#mrd_updated_time,mrd_deleted_time = :#mrd_deleted_time,updated_time = NOW() AT TIME ZONE 'utc'?dataSource=#dataSource
```
[the remark this answers is not in the document]
Comment (kiran-yenigala-hmcts) [reply src/main/resources/application-lrd-building-location-router.yaml:14]: batch=true no longer works with the query written as is, we have to rewrite the query to make it work

File: src/main/resources/application-lrd-court-venue-router.yaml [hunk 5]
```diff
@@ -27,9 +27,9 @@ route:
       welsh_court_name = :#welsh_court_name, uprn = :#uprn, venue_ou_code = :#venue_ou_code, mrd_building_location_id = :#mrd_building_location_id,
       mrd_venue_id = :#mrd_venue_id, service_url = :#service_url, fact_url = :#fact_url,
       mrd_created_time = :#mrd_created_time, mrd_updated_time = :#mrd_updated_time, mrd_deleted_time = :#mrd_deleted_time,
-      external_short_name = :#external_short_name, updated_time = NOW() AT TIME ZONE 'utc'?batch=true&dataSource=dataSource
+      external_short_name = :#external_short_name, updated_time = NOW() AT TIME ZONE 'utc'?dataSource=#dataSource
```
[the remark this answers is not in the document]
Comment (kiran-yenigala-hmcts) [reply src/main/resources/application-lrd-court-venue-router.yaml:30]: batch=true no longer works with the query written as is, we have to rewrite the query to make it work

File: src/functionalTest/resources/application-integration.yml [hunk 6]
```diff
@@ -76,9 +76,9 @@ route:
...
       ON CONFLICT (epimms_id) DO UPDATE SET building_location_name = :#building_location_name,building_location_status = :#building_location_status,area = :#area,region_id = :#region_id,cluster_id = :#cluster_id,
-      court_finder_url = :#court_finder_url,postcode = :#postcode,address = :#address,welsh_building_location_name = :#welsh_building_location_name,welsh_address = :#welsh_address,uprn = :#uprn,latitude = :#latitude,longitude = :#longitude,mrd_building_location_id = :#mrd_building_location_id,mrd_created_time = :#mrd_created_time,mrd_updated_time = :#mrd_updated_time,mrd_deleted_time = :#mrd_deleted_time,updated_time = NOW() AT TIME ZONE 'utc'?batch=true&dataSource=dataSource
+      court_finder_url = :#court_finder_url,postcode = :#postcode,address = :#address,welsh_building_location_name = :#welsh_building_location_name,welsh_address = :#welsh_address,uprn = :#uprn,latitude = :#latitude,longitude = :#longitude,mrd_building_location_id = :#mrd_building_location_id,mrd_created_time = :#mrd_created_time,mrd_updated_time = :#mrd_updated_time,mrd_deleted_time = :#mrd_deleted_time,updated_time = NOW() AT TIME ZONE 'utc'?dataSource=#dataSource
```
[the remark this answers is not in the document]
Comment (kiran-yenigala-hmcts) [reply src/functionalTest/resources/application-integration.yml:79]: batch=true no longer works with the query written as is, we have to rewrite the query to make it work

**Pull request merged**